util/network/network.go

```diff
@@ -2,19 +2,51 @@ package network
 
 import (
 	"io"
+	"os"
 
 	"github.com/moby/buildkit/solver/pb"
 	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
 )
 
-// Default returns the default network provider set
-func Default() map[pb.NetMode]Provider {
+type Opt struct {
+	Root          string
+	Mode          string
+	CNIConfigPath string
+	CNIBinaryDir  string
+}
+
+// Providers returns the network provider set
+func Providers(opt Opt) (map[pb.NetMode]Provider, error) {
+	var defaultProvider Provider
+	switch opt.Mode {
+	case "cni":
+		cniProvider, err := NewCNIProvider(opt)
+		if err != nil {
+			return nil, err
+		}
+		defaultProvider = cniProvider
+	case "host":
+		defaultProvider = NewHostProvider()
+	case "auto", "":
```

Comment:
Not huge fan of "auto". At least, it should print warning when falling back to host network, as it seems less secure unless already containerized

Reply:
https://github.com/moby/buildkit/pull/309#discussion_r193636050 :)

Reply:
😱 oh sorry

Reply:
added warning